firmware/signing.c

```diff
@@ -777,10 +786,6 @@ void signing_txack(TransactionType *tx)
 			signing_check_input(&tx->inputs[0]);
 			if (tx->inputs[0].script_type == InputScriptType_SPENDMULTISIG
 				|| tx->inputs[0].script_type == InputScriptType_SPENDADDRESS) {
-				// remember the first non-segwit input -- this is the first input
```

Comment:
Don't make bcc inputs nonsegwit_inputs.  This way they use the segwit path that avoids serializing the whole transaction again.   This code moved below in the non-forkid path.